modules/dataframe/src/io/csvreader.cpp

```diff
@@ -510,6 +524,25 @@ std::shared_ptr<DataFrame> CSVReader::readData(std::istream& stream) const {
         std::remove_if(rows.begin(), rows.end(), [](const auto& row) { return row.first.empty(); }),
         rows.end());
 
+    // Ignore comment lines 
+    // The row must start with the comment character. Applied before keep only
+    if (!rowComment_.empty()) {
+        rows.erase(std::remove_if(rows.begin(), rows.end(),
+                                  [this](const auto& row) {
+                                      return row.first.substr(0, rowComment_.size()) == rowComment_;
```

Comment:
yes, that should work, both this and the empty line check could be there. probably we want to call out to a
` bool filter(std::string_view line)` function there. 
Then the `filter` function can decide what to include/exclude. 

One might have to consider if the Header row is affected by all this somehow?
We might want to apply some filter after the header row is removed.

Reply:
Sounds reasonable to me.

Re:header: We could include a firstRowHeader-check by line 519 to see if it should be kept. Otherwise i guess the filtering should apply to the first row as well?